### ISSUE TITLE
Incompatibility of ioTimer in Cats interop

### ISSUE THREAD

**Issue by ghostdogpr** [scalaz/scalaz-zio#587]
Cats' `ConcurrentEffect[F]` requires a `Timer[F]`.
There was in our catz interop an `implicit def ioTimer` that returned a `Timer[Task]`.

With the new ZIO Environment, ioTimer is defined as below:
```scala
implicit def ioTimer[R <: Clock, E]: effect.Timer[ZIO[R, E, ?]] = new effect.Timer[ZIO[R, E, ?]]
```

It returns a `Timer[ZIO[R, E, ?]` with `R <: Clock`, but this doesn't match the definition of `Task`.

Shall we use `Clock.Live` directly instead of relying on an environment here? Or require an implicit parameter `Clock` as it was before? Any other suggestion?

**Comment by kaishh** [comment]
It should probably just take an implicit Clock as before and use IO.provide to discharge back to Task. We can provide a low priority instance for ZIO[R] without an implicit too

**Comment by ghostdogpr** [comment]
Sounds good. I’ll work on this tomorrow.

**Comment by ghostdogpr** [comment]
Closed by #593
